### PR TITLE
First revision of the Agari connector

### PR DESCRIPTION
**Pull request by udinagari**
Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: DataConnectors/Agari/AzureFunctionAgari/run.ps1 [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+        $sgapi_uri = 'https://graph.microsoft.com/beta/security/tiIndicators'
+        $sgapi_headers = @{
+                            'Authorization' = "Bearer $sgapi_token"
+                            'ContentType' = 'application/json'
+                        }
+}
+
+#Set uris for get/revoke bearer tokens
+$get_apd_token_uri='https://api.agari.com/v1/ep/token'
+$get_bp_token_uri='https://api.agari.com:443/v1/cp/oauth/token'
```
Comment (djspencer) [inline DataConnectors/Agari/AzureFunctionAgari/run.ps1]: For consistency you can drop :443

File: DataConnectors/Agari/AzureFunctionAgari/run.ps1 [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+# --------------------------------------------------- #
+# Below is for loading BP Logs and TIs into Sentinel. #
+# --------------------------------------------------- #
+
+#Check if BP is enabled and we got a token
+if (($bpEnabled) -and ($bpToken)) {
+    $offset = 0
+    do {
+        $BPAlertsListAPI = "https://api.agari.com/v1/cp/alert_events?start_date=$startdate&end_date=$enddate&fields=id%2C&limit=$limit&offset=$offset&sort=created_at%20ASC"
```
Comment (djspencer) [inline DataConnectors/Agari/AzureFunctionAgari/run.ps1:234]: Nice use of &fields here to reduce payload size

**Review by djspencer — COMMENTED** [review]
Just a couple inline comments.
I don't know Sentinel, but the code looks pretty clean from what I can tell.
I appreciate the comments!

One thing we will want to do is add commit comments along with the JIRA card reference before moving this forward.

File: DataConnectors/Agari/AzureFunctionAgari/run.ps1 (hunk 2 above)
Comment (udinagari) [reply DataConnectors/Agari/AzureFunctionAgari/run.ps1:234]: Thanks!  If you got the user-agent string I can add it to the web-requests.

Comment (djspencer) [reply DataConnectors/Agari/AzureFunctionAgari/run.ps1:234]: Doh, forgot that.
"AgariSentinel BP_APD_Integration/v1.0 MicrosoftSentinel/v?.?"

Any chance you can get the hosting Sentinel version programmatically?

**Comment by udinagari** [comment]
I'll check if there is a version code. We could also use the client id if we want to identify the Agari customer.
________________________________

**Pull request closed without merging**